app/models/manageiq/providers/openstack/network_manager/cloud_subnet.rb

```diff
@@ -121,7 +126,10 @@ def self.params_for_create(ems)
   end
 
   def self.raw_create_cloud_subnet(ext_management_system, options)
-    cloud_tenant = options.delete(:cloud_tenant)
+    cloud_network = CloudNetwork.find_by(:id => options[:cloud_network_id])
```

Comment:
Since `options` is passed in as-is to `subnet = service.subnets.new(options)` below we should delete this from the options hash.

Best to follow the same pattern that you have for the cloud tenant e.g.:
```
cloud_network_id = options.delete(:cloud_network_id)
cloud_network = CloudNetwork.find_by(:id => cloud_network_id) if cloud_network_id
options[:network_id] = cloud_network&.ems_ref
```